### PR TITLE
Fix ansible setup playbook

### PR DESCRIPTION
**Pull request by lehoanganh**
* only use docker-machine commands in mac env. add db host on-the-fly to ansible runtime once mac env is used
* improve README.md

### CODE REVIEW

**Comment by lehoanganh** [comment]
```
lha@dyn-9-152-96-137  ~/Development/hoang/openwhisk/ansible   fix_ansible_setup  ansible-playbook -i environments/mac setup.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
Friday 17 June 2016  13:15:49 +0200 (0:00:00.016)       0:00:00.016 ***********
ok: [localhost]

TASK [find the ip of docker-machine] *******************************************
Friday 17 June 2016  13:15:49 +0200 (0:00:00.260)       0:00:00.277 ***********
changed: [localhost -> localhost]

TASK [get the docker-machine ip] ***********************************************
Friday 17 June 2016  13:15:49 +0200 (0:00:00.298)       0:00:00.576 ***********
ok: [localhost]

TASK [gen hosts for docker-machine] ********************************************
Friday 17 June 2016  13:15:50 +0200 (0:00:00.040)       0:00:00.616 ***********
changed: [localhost -> localhost]

TASK [add new db host on-the-fly] **********************************************
Friday 17 June 2016  13:15:50 +0200 (0:00:00.200)       0:00:00.817 ***********
changed: [localhost]

TASK [prepare db_local.ini] ****************************************************
Friday 17 June 2016  13:15:50 +0200 (0:00:00.041)       0:00:00.858 ***********
ok: [localhost -> localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=6    changed=3    unreachable=0    failed=0

Friday 17 June 2016  13:15:50 +0200 (0:00:00.203)       0:00:01.062 ***********
===============================================================================
TASK: find the ip of docker-machine ------------------------------------- 0.30s
TASK: setup ------------------------------------------------------------- 0.26s
TASK: prepare db_local.ini ---------------------------------------------- 0.20s
TASK: gen hosts for docker-machine -------------------------------------- 0.20s
TASK: add new db host on-the-fly ---------------------------------------- 0.04s
TASK: get the docker-machine ip ----------------------------------------- 0.04s
```

**Comment by lehoanganh** [comment]
```
lha@dyn-9-152-96-137  ~/Development/hoang/openwhisk/ansible   fix_ansible_setup  ansible-playbook -i environments/local setup.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.017)       0:00:00.017 ***********
ok: [localhost]

TASK [find the ip of docker-machine] *******************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.263)       0:00:00.281 ***********
skipping: [localhost]

TASK [get the docker-machine ip] ***********************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.037)       0:00:00.318 ***********
skipping: [localhost]

TASK [gen hosts for docker-machine] ********************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.037)       0:00:00.356 ***********
skipping: [localhost]

TASK [add new db host on-the-fly] **********************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.033)       0:00:00.390 ***********
skipping: [localhost]

TASK [prepare db_local.ini] ****************************************************
Friday 17 June 2016  13:16:25 +0200 (0:00:00.034)       0:00:00.425 ***********
ok: [localhost -> localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0

Friday 17 June 2016  13:16:26 +0200 (0:00:00.197)       0:00:00.622 ***********
===============================================================================
TASK: setup ------------------------------------------------------------- 0.26s
TASK: prepare db_local.ini ---------------------------------------------- 0.20s
TASK: find the ip of docker-machine ------------------------------------- 0.04s
TASK: get the docker-machine ip ----------------------------------------- 0.04s
TASK: add new db host on-the-fly ---------------------------------------- 0.03s
TASK: gen hosts for docker-machine -------------------------------------- 0.03s
```

File: ansible/setup.yml [hunk 1]
```diff
@@ -6,14 +6,20 @@
   - name: find the ip of docker-machine
     local_action: shell "docker-machine" "ip" "{{docker_machine_name | default('whisk')}}"
     register: result
-    ignore_errors: true
+    when: "'environments/mac' in inventory_dir"
```
Comment (markusthoemmes) [inline ansible/setup.yml]: I'ld rather not hardcode an environment here. We had that in ant before, problems and strange behaviour inevitable. Couldn't we just check if `docker-machine` is available as a command?

Comment (lehoanganh) [inline ansible/setup.yml]: currently we support only the 2 env `local` or `mac`. Users have to input `-i environments/local` or `-i environments/mac`. For now, `setup.yml` is strongly bind to a env, cause it needs to generate a `db_local.ini` based on template.

Another checking `docker-machine` as a command always gives a non-zero return code which also creates a "fatal" error in ansible view -> again, have to ignore this.

Comment (domdom82) [inline ansible/setup.yml]: we could experiment with 
```
which docker-machine ; exit 0
```

Comment (lehoanganh) [inline ansible/setup.yml]: didnt work. Can not find out how.

Comment (lehoanganh) [inline ansible/setup.yml]: if we can merge `local` and `mac` such 'cheating' extra work is not needed anymore ;)

Comment (rabbah) [inline ansible/setup.yml]: docker machine nfs might work.

Comment (lehoanganh) [inline ansible/setup.yml]: related to markus' concern, if user want to create a new own mac env, it also works fine cause the ips are input in the env. Our `mac` env uses 'dynamic' IP which are generated by setup.yml. So, the `when` condition is there for good reasons.

**Comment by lehoanganh** [comment]
pull request closed. will open a new one once I set up my own branch

**Comment by csantanapr** [comment]
@lehoanganh what's the new PR #?

**Pull request closed without merging**